### ISSUE TITLE
add an option to disable the auto-highlighting for current word when choose a snippet

### ISSUE THREAD

**Issue by hefei00** [Microsoft/vscode#56880]
after update the VSCode 1.26, I find it is frustrating when choosing a snippet in vscode.

Like the gif below, after choosing the import snippet, the current word was selected and highlight, so the intellisense is ineffective. I must press ESC key to cancel the selection to enable the intellisense. And there are other snippets that have the similar action. 

Is there any way to disable the selection after choosing an snippet?

https://i.stack.imgur.com/F5vqT.gif

**Comment by hefei00** [comment]
@jrieken thanks for your responses! but the setting:`"editor.suggest.snippetsPreventQuickSuggestions": false` does not work. I think it is nothing about the snippet mode prevent the quick suggestions. It is about the snippet it self. 

like the snippets below.  when I enter `todo` and choose the TODO snippet, the current token/word will be selected. when I enter `fix` and choose the FIXME snippet, the current token/word will not be selected. The difference between two snippets are the `$1` word. and if the token/word is selected and highlighted, the quick suggestion will not work. 

```json
	"todo": {
		"prefix": "todo",
		"scope": "javascript,typescript",
		"body": "// TODO: $1"
	},
	"fix": {
		"prefix": "fix",
		"scope": "javascript,typescript",
		"body": "// FIXME: "
	},
```

**Comment by hefei00** [comment]
I ask a question an stackoverflow too. https://stackoverflow.com/questions/51932709/how-to-disable-auto-highlighting-for-the-current-word-in-vscode

**Comment by jrieken** [comment]
> TODO snippet, the current token/word will be selected. when I enter fix and choose the FIXME snippet, the current token/word will not be 

Sorry, but that's not the selection but the highlight for 'typing inside a snippet placeholder'. The `fix`-snippet doesn't use the `$1` placeholder and therefore typing at its end will not expand/grow a placeholder anything.

**Comment by hefei00** [comment]
@jrieken thanks. Is the highlight for "typing inside a snippet placeholder" a feature by design? what can I do to change this? It seems that the snippets are provided by the javascript language basic. But I don't know how to change the default behaviour.

**Comment by jrieken** [comment]
Yes - that's the design and also nothing specific to VS Code, all editors I know have some kind of highlight for placeholders

**Comment by hefei00** [comment]
got it! thanks for your explanation.
